metadata-io/src/main/java/com/linkedin/metadata/graph/neo4j/Neo4jGraphService.java

```diff
@@ -90,39 +100,105 @@ public void addEdge(@Nonnull final Edge edge) {
+    if (edge.getCreatedOn() != null) {
+      setCreatedOnTemplate = "r.createdOn = " + edge.getCreatedOn();
+      propertiesTemplateJoiner.add(setCreatedOnTemplate);
+    }
+    if (edge.getCreatedActor() != null) {
+      setcreatedActorTemplate = "r.createdActor = '" + edge.getCreatedActor() + "'";
+      propertiesTemplateJoiner.add(setcreatedActorTemplate);
+    }
+    if (edge.getUpdatedOn() != null) {
+      setupdatedOnTemplate = "r.updatedOn = " + edge.getUpdatedOn();
+      propertiesTemplateJoiner.add(setupdatedOnTemplate);
+    }
+    if (edge.getUpdatedActor() != null) {
+      setupdatedActorTemplate = "r.updatedActor = '" + edge.getUpdatedActor() + "'";
+      propertiesTemplateJoiner.add(setupdatedActorTemplate);
+    }
+    if (edge.getProperties() != null) {
+      for (Map.Entry<String, Object> entry : edge.getProperties().entrySet()) {
+        if (entry.getValue() instanceof String) {
+          setPropertyTemplate = "r." + entry.getKey() + " = '" + entry.getValue() + "'";
+          propertiesTemplateJoiner.add(setPropertyTemplate);
+        } else {
+          throw new UnsupportedOperationException(String.format(
+              "Tried setting properties on graph edge but property value type is not supported. Key: %s, Value: %s ",
+              entry.getKey(), entry.getValue()));
+        }
+      }
+    }
+    if (!StringUtils.isEmpty(propertiesTemplateJoiner.toString())) {
+      statement = statement + "SET " + propertiesTemplateJoiner;
+    }
 
-    statements.add(buildStatement(statement, paramsMerge));
+    statements.add(buildStatement(statement, new HashMap<>()));
 
     executeStatements(statements);
   }
 
   @Override
   public void upsertEdge(final Edge edge) {
-    throw new UnsupportedOperationException("Upsert edge not supported by Neo4JGraphService at this time.");
+    addEdge(edge);
   }
 
   @Override
   public void removeEdge(final Edge edge) {
-    throw new UnsupportedOperationException("Remove edge not supported by Neo4JGraphService at this time.");
+    log.debug(
+        String.format("Deleting Edge source: %s, destination: %s, type: %s", edge.getSource(), edge.getDestination(),
+            edge.getRelationshipType()));
+
+    final String sourceType = edge.getSource().getEntityType();
+    final String destinationType = edge.getDestination().getEntityType();
+
+    final List<Statement> statements = new ArrayList<>();
+
+    // DELETE relationship
+    final String mergeRelationshipTemplate =
+        "MATCH (source:%s {urn: $sourceUrn})-[r:%s]->(destination:%s {urn: $destinationUrn}) DELETE r";
+    final String statement =
+        String.format(mergeRelationshipTemplate, sourceType, edge.getRelationshipType(), destinationType);
+
+    final Map<String, Object> paramsMerge = new HashMap<>();
+    paramsMerge.put("sourceUrn", edge.getSource().toString());
+    paramsMerge.put("destinationUrn", edge.getDestination().toString());
```

Comment:
just curious - is there any difference or benefit in using `$sourceUrn` and `paramsMerge.put("sourceUrn"...` as opposed to just inserting it in the template like you're doing above and for other things like `sourceType`?

Reply:
right, i will make utils file and simplify logic. hope i could make a new commit today